firstvoices/backend/views/custom_order_recalculate_views.py

```diff
@@ -67,21 +68,26 @@ def create(self, request, *args, **kwargs):
 
         # Call the recalculation task
         try:
-            recalculation_results = recalculate_custom_order.apply_async((site_slug,))
-
-            # Delete any previous results
-            CustomOrderRecalculationResult.objects.filter(
+            # Check if preview task for the same site is ongoing
+            previous_tasks = CustomOrderRecalculationResult.objects.filter(
                 site=site[0], is_preview=False
-            ).delete()
-
-            # Save the result to the database
-            CustomOrderRecalculationResult.objects.create(
-                site=site[0],
-                latest_recalculation_result=recalculation_results.get(timeout=360),
-                task_id=recalculation_results.task_id,
-                is_preview=False,
             )
-
+            running_tasks = 0
+            if len(previous_tasks) > 0:
+                for task in previous_tasks:
+                    status = AsyncResult(task.task_id).status
+                    if status == "PENDING":
+                        running_tasks += 1
+
+            if running_tasks > 0:
+                return Response(
+                    {
+                        "message": "Recalculation is already running. Please use retrieve API to verify status."
+                    },
+                    status=202,
```

Comment:
I think this is a good workflow, but this is not the right status code for this type of response. If we want to tell someone to GET another URL instead of processing this POST, we should return status 303 (see other) and put the destination URL in the Location header. https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections